internal/metrics/kafka.go

```diff
@@ -0,0 +1,80 @@
+package metrics
+
+import (
+	"github.com/etf1/kafka-mongo-watcher/internal/kafka"
+	"github.com/prometheus/client_golang/prometheus"
+	kafkaconfluent "gopkg.in/confluentinc/confluent-kafka-go.v1/kafka"
+)
+
+var metricsRegistered = false
+
+type KafkaRecorder struct {
+	producerSuccessCounter *prometheus.CounterVec
+	producerErrorCounter   *prometheus.CounterVec
+}
+
+// NewKafkaRecorder returns a kafka recorder that is used to send metrics
+func NewKafkaRecorder() *KafkaRecorder {
+	return &KafkaRecorder{
+		// Kafka producer metrics
+		producerSuccessCounter: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "kafka",
+				Name:      "producer_success_counter_total",
+				Help:      "This represent the number of successful messages pushed into Kafka",
+			},
+			[]string{"topic"},
+		),
+		producerErrorCounter: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "kafka",
+				Name:      "producer_error_counter_total",
+				Help:      "This represent the number of error messages handled by Kafka producer",
+			},
+			[]string{"topic"},
+		),
+	}
+}
+
+// RegisterOn allows to specify a specific Prometheus registry
+func (r *KafkaRecorder) RegisterOn(registry prometheus.Registerer) *KafkaRecorder {
+	if registry == nil {
+		registry = prometheus.DefaultRegisterer
+	}
+
+	if !metricsRegistered {
```

Comment:
I added this check for integration tests because the registry always have the metrics registered when launching a second test function but I agree it's not clean, I will add a `Unregister()` method instead and unregister them at the end of test function.